consensus/block_validation.go

```diff
@@ -0,0 +1,87 @@
+package consensus
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+// BlockValidator defines an interface used to validate a blocks syntax and
+// semantics.
+type BlockValidator interface {
+	BlockSemanticValidator
+	BlockSyntaxValidator
+}
+
+// BlockSemanticValidator defines an interface used to validate a blocks
+// semantics.
+type BlockSemanticValidator interface {
+	ValidateSemantic(ctx context.Context, child, parent *types.Block) error
+}
+
+// BlockSyntaxValidator defines an interface used to validate a blocks
+// syntax.
+type BlockSyntaxValidator interface {
```

Comment:
nit: I don't see the value in splitting these interfaces up here instead of defining a single `BlockValidator` interface with the two functions.  But not a strong opinion.

Reply:
One advantage is that we can pass just the syntax validator to methods that need it, e.g:
```go
NewFetcher(ctx context.Context, bsrv bserv.BlockService, bv consensus.BlockSyntaxValidator) *Fetcher
```
There are other ways to accomplish this -- like defining a sub-interface in the `fetcher.go` file. As work in this areas continues I am opening to changing these interfaces as needed, for now will leave as is.

Reply:
I think splitting is right. The interface definitions should move to the consumers as this matures.